hibernate-core/src/test/java/org/hibernate/test/collection/delayedOperation/DetachedBagDelayedOperationTest.java

```diff
@@ -0,0 +1,387 @@
+				this::sessionFactory, session -> {
+					Parent p = session.get( Parent.class, 1L );
+					assertFalse( Hibernate.isInitialized( p.getChildren() ) );
+					// initialize
+					Hibernate.initialize( p.getChildren() );
+					assertTrue( Hibernate.isInitialized( p.getChildren() ) );
+					return p;
+				}
+		);
+		final Parent pAfterDetachWithQueuedOperations = doInHibernate(
+				this::sessionFactory, session -> {
+					Parent p = (Parent) session.merge( pOriginal );
+					Child c = new Child( "Zeke" );
+					c.setParent( p );
+					session.persist( c );
+					assertFalse( Hibernate.isInitialized( p.getChildren() ) );
+					p.getChildren().add( c );
+					assertFalse( Hibernate.isInitialized( p.getChildren() ) );
+					assertTrue( ( (AbstractPersistentCollection) p.getChildren() ).hasQueuedOperations() );
+
+					checkTriggerablesNotTriggered();
+					session.detach( p );
+					assertTrue( triggerableQueuedOperationWhenDetachFromSession.wasTriggered() );
+					assertEquals(
+							"HHH000496: Detaching an uninitialized collection with queued operations from a session: [org.hibernate.test.collection.delayedOperation.DetachedBagDelayedOperationTest$Parent.children#1]",
+							triggerableQueuedOperationWhenDetachFromSession.triggerMessage()
+					);
+					triggerableQueuedOperationWhenDetachFromSession.reset();
+
+					// Make sure nothing else got triggered
+					checkTriggerablesNotTriggered();
+
+					return p;
+				}
+		);
+
+		checkTriggerablesNotTriggered();
+
+		assertTrue( ( (AbstractPersistentCollection) pAfterDetachWithQueuedOperations.getChildren() ).hasQueuedOperations() );
+
+		// Save detached Parent with uninitialized collection with queued operations
+		doInHibernate(
+				this::sessionFactory, session -> {
+
+					checkTriggerablesNotTriggered();
+
+					assertFalse( triggerableQueuedOperationWhenAttachToSession.wasTriggered() );
+					session.saveOrUpdate( pAfterDetachWithQueuedOperations );
+					assertTrue( triggerableQueuedOperationWhenAttachToSession.wasTriggered() );
+					assertEquals(
+							"HHH000495: Attaching an uninitialized collection with queued operations to a session: [org.hibernate.test.collection.delayedOperation.DetachedBagDelayedOperationTest$Parent.children#1]",
+							triggerableQueuedOperationWhenAttachToSession.triggerMessage()
+					);
+					triggerableQueuedOperationWhenAttachToSession.reset();
+
+					// Make sure nothing else got triggered
+					checkTriggerablesNotTriggered();
+
+					assertFalse( Hibernate.isInitialized( pAfterDetachWithQueuedOperations.getChildren() ) );
+					assertTrue( ( (AbstractPersistentCollection) pAfterDetachWithQueuedOperations.getChildren() ).hasQueuedOperations() );
+
+					// Queued operations will be executed when the collection is initialized,
+					// After initialization, the collection will contain the Child that was added as a
+					// queued operation before being detached above.
+					Hibernate.initialize( pAfterDetachWithQueuedOperations.getChildren() );
```

Comment:
When initialized, the collection **contains** the Child that was added to the collection by the queued operations before being detached. That's because `pAfterDetachWithQueuedOperations` was attached to the session  using `Session#saveOrUpdate`.